libraries/rush-lib/src/cli/RushPnpmCommandLineParser.ts

```diff
@@ -281,6 +281,14 @@ export class RushPnpmCommandLineParser {
             );
             throw new AlreadyReportedError();
           }
+          // patch-commit internally calls installation under cwd instead of the common/temp folder
+          // It throws missing package.json error, so in this case, we need to set the dir to the common/temp folder here
+          if (pnpmArgs.indexOf('--dir') < 0 && pnpmArgs.indexOf('-C') < 0) {
```

Comment:
Is `-C` just the short form for `--dir`?

Reply:
Seems like we should error in this case and tell people not to pass `--dir` or `-C` if they're running `patch-commit`.

Reply:
I suppose it doesn't harm people to pass -C if they want to do something very specific, but if it doesn't exist, we should add the flag to point at our temp folder.

The same logic applies to commands like "rush-pnpm why", which doesn't work right unless in temp folder.

Reply:
> Is -C just the short form for --dir?

Yes.

> I suppose it doesn't harm people to pass -C if they want to do something very specific, but if it doesn't exist, we should add the flag to point at our temp folder.

Yes. The user story would be

1. Run `rush-pnpm patch-commit <path>` under the repo root works ✅
2. Run `rush-pnpm patch-commit <path>` under a project folder works ✅
3. Run `rush-pnpm patch-commit <path> --dir <specified_dir>` works ✅